src/data/fin.lean

```diff
@@ -415,10 +431,47 @@ order_embedding.of_strict_mono (λ i, ⟨n + (i : ℕ), add_lt_add_left i.2 _⟩
 
 @[simp] lemma coe_nat_add (n : ℕ) {m : ℕ} (i : fin m) : (nat_add n i : ℕ) = n + i := rfl
 
-theorem le_last (i : fin (n+1)) : i ≤ last n :=
-le_of_lt_succ i.is_lt
+@[simp] lemma coe_order_iso_apply (e : fin n ≃o fin m) (i : fin n) : (e i : ℕ) = i :=
```

Comment:
This lemma strikes me as odd because `(e : fin n ≃o fin m)` can only exist if `n = m`. Perhaps it's still convenient to write it this way though. Would a comment like this make sense?
```suggestion
/-- Note that implicitly this requires that `n = m` -/
@[simp] lemma coe_order_iso_apply (e : fin n ≃o fin m) (i : fin n) : (e i : ℕ) = i :=
```